src/workerd/api/global-scope.c++

```diff
@@ -601,7 +601,7 @@ jsg::Promise<jsg::Ref<Response>> ServiceWorkerGlobalScope::fetch(
 jsg::Ref<Socket> ServiceWorkerGlobalScope::connect(
     jsg::Lock& js, kj::String address, jsg::Optional<SocketOptions> options,
     CompatibilityFlags::Reader featureFlags) {
-  return connectImpl(js, nullptr, kj::mv(address), featureFlags);
+  return connectImpl(js, nullptr, kj::mv(address), options, featureFlags);
```

Comment:
Nit: While the struct is trivially copiable now and this works, I wonder if we shouldn't go ahead and `kj::mv(options)` here now in case that changes in the future? Non-blocking tho